src/store/state/board_instance_add.go

```diff
@@ -81,7 +81,28 @@ func (bi *BoardInstance) NewPost(post *r0.Post) (uint64, error) {
 		// Add post to board page.
 		tpRef, tPage, e := pages.BoardPage.GetThreadPage(tpHash, nil)
 		if e != nil {
-			return e
+			// TODO: Fix bug. Workaround...
+			// <<< START : WORKAROUND >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
+
+			var doneError = errors.New("done")
+			e := pages.BoardPage.Threads.Ascend(func(i int, tpRefGot *skyobject.RefsElem) (err error) {
+				if tpRefGot.Hash == tpHash {
+					tpRef = tpRefGot
+					tpValue, e := tpRef.Value()
+					if e != nil {
+						return e
+					}
+					tPage, _ = tpValue.(*r0.ThreadPage)
```

Comment:
Has nothing to do with the problem. Can ignore

---

By design, this must work
```go
tPage = tpValue.(*ro.ThreadPage) // without the underscore
```
Feel free to omit the second (ok) argument for Ref. If it panics, then it is bug of CXO.
If some Ref represents nil, then it return nil-pointer of the type.

But the Refs must not return any element that represents nil. It treats all elements with `hash == (cipher.SHA256{})` as deleted. Thus, you never get the `tpRefGot` that represents nil.